### PR TITLE
Initial state with redux

### PR DESCRIPTION
**Pull request by jossefaz**
* init API registry + Mock server call
* initial state in redux
* First selector for anouncing torrents

### CODE REVIEW

**Comment by anthonyraymond** [comment]
Hello buddy 😃 

i have a problem with the project, it does not "compile".

>./src/store/reducers/index.js
>Module not found: Can't resolve './client' in 'D:\workspace\node\owl-webui\src\store\reducers'


And also,
i can't run the project on Windows using `yarn start`, i need to alter the package.json:
image

**Comment by jossefaz** [comment]
I used npm....not yarn....so npm start should work

**Comment by jossefaz** [comment]
I just added the missing folder...sorry for that. So just 
` npm i && npm start ` 
and everything should be OK

File: src/store/reducers/bandwidth/index.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import types from "../../actions/types";
+
+const reducer = (state = {}, action) => {
+    debugger
```
Comment (anthonyraymond) [inline src/store/reducers/bandwidth/index.js]: woops ^^

File: src/store/reducers/client/index.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import types from "../../actions/types";
+
+const reducer = (state = {}, action) => {
+    debugger
```
Comment (anthonyraymond) [inline src/store/reducers/client/index.js]: woops ^^

File: src/store/reducers/config/index.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import types from "../../actions/types";
+
+const reducer = (state = {}, action) => {
+    debugger
```
Comment (anthonyraymond) [inline src/store/reducers/config/index.js]: woops ^^

**Review by anthonyraymond — COMMENTED** [review]
i've added an .env file and remove the variable in react start which prevent start on windows environment

**Comment by anthonyraymond** [comment]
Can you add the config/config.json file ? So we can start developing without diging into the source code to guess the fiel format etc..


Also, why ECMAScript instead of typescript?


Other than that looks good

File: src/store/reducers/config/index.js (hunk 3 above)
Comment (jossefaz) [reply src/store/reducers/config/index.js]: removed

File: src/store/reducers/client/index.js (hunk 2 above)
Comment (jossefaz) [reply src/store/reducers/client/index.js]: removed

File: src/store/reducers/bandwidth/index.js (hunk 1 above)
Comment (jossefaz) [reply src/store/reducers/bandwidth/index.js]: removed

**Review by jossefaz — COMMENTED** [review]
The config/config.json is added yet (since the first PR) : it is in the public folder (since it is the root folder of the application).
I fixed the other ("debuggers" in reducers).
I wrote in ECMA because I did not began to work with TS yet at my work so I do not really know all the stuff (I did some project with it but this was two years ago and since then I wrote all my projects in ECMA much faster to wrtie even if its not type safe)
Now if you really want TS tell me and I will convert my code to TS but it could take a while...

**Comment by anthonyraymond** [comment]
I'm not dure to understand why the env variable has this value then.
Howbcould that work? Env value wont be available through proccess.env when build and in production environnent. Even if if would the ./config/config.json will sure never resolve either.

Ok for es6, i will do the migration to ts later on

**Comment by jossefaz** [comment]
Of course it will be available.
Webpack chunks are all available from the root folder. Then this value is a relative path to that folder. 

But not matters : that what we discuss together : fornow (for the development) it is based on a stand alone file inside the web-ui code base (in the public folder). But in future there would not be any env at all since the webui will receive this file by an api call. It then will be the only hard coded adress in the code which will be the route to the server that will send to the client the config.json file.

Indeed, environement variable are not available for change after webpack build

**Comment by anthonyraymond** [comment]
Huuum i might have forgot about that 😄.
Ok i get it now. But the path still looks wrong, shouldn't it be ./public/config/config.json?

Le 10 janv. 2021 à 11:59, à 11:59, jossefaz <notifications@github.com> a écrit:

**Comment by jossefaz** [comment]
No...Public is the root folder...all chunkcs are loaded inside of it (at least that's how it works for me...I am not a webpack freak but that's how I understand it)

**Comment by anthonyraymond** [comment]
Ah ! ok i get it.

Since the `REACT_APP_CONFIG_PATH=./config/config.json` was starting with a `.` i though that the dev version was fetched using hte file system (which did not made sense at all in the context of a webapp.
But it's actually reading a file exposed by himself, ok my bad.

To make it a bit more clear can you remove the "." before the path in the `.env` file?

**Comment by jossefaz** [comment]
removed

**Comment by anthonyraymond** [comment]
Ok, now that my brain has booted up i understand how it works 🤣 

Merged :)

**Pull request merged**